bbc/digital-paper-edit-firebase
# Add Python server

**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
https://github.com/bbc/digital-paper-edit-infrastructure/issues/21 

**Describe what the PR does**    
- Add a simple Python server for use on the deployed DPE EC2. Works in conjunction with this [PR](https://github.com/bbc/digital-paper-edit-infrastructure/pull/22) in the infrastructure project. It has a /status healthcheck endpoint for AWS use, and all other requests fall back to /index.html where the React app is served from.
- Adds an isProduction() function to check whether Dev/Test or Prod/Live config should be used (this is added through the build process in the digital-paper-edit-infrastructure project).
- Changes some `const` variables to `let`where reassignment was causing read-only errors when deployed

**State whether the PR is ready for review or whether it needs extra work**    
RFR

**How to test**
1. `npm run build`
2. `cd build`
3. `python server.py`
4. Go to http://localhost:8080
5. Check the app name appears in the page header (this shows that env variables are working correctly - ask Tamsin or Alli for the updated config)
6. Log in and create a new project, update the name of a project, and add a new programme script to a project (this shows the read-only errors are fixed)
